semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/http/HttpSemanticConventionUtils.java

```diff
@@ -196,4 +257,257 @@ public static boolean isValidUrl(String url) {
   public static URL getNormalizedUrl(String url) throws MalformedURLException {
     return new URL(new URL(RELATIVE_URL_CONTEXT), url);
   }
+
+  public static Optional<String> getHttpUserAgent(Event event) {
+
+    if (event.getAttributes() == null || event.getAttributes().getAttributeMap() == null) {
```

Comment:
A lot of methods in this class are just doing the task of finding the first attribute in the map, can you just use org.hypertrace.core.datamodel.shared.SpanAttributeUtils#getFirstAvailableStringAttribute

Reply:
Applies for method in the RpcSemanticConventionUtils class

Reply:
But it first searches in enriched attributes and if doesn't find over there then searches in the attributes
Will that be a problem ?

Reply:
No that shouldn't be a problem, since enriched attribute keys are different from raw attributes. 
Also, we do use this method to fetch raw-attributes in existing code

Reply:
Replaced. But some functions had additional checks so kept them as it is